src/compiler/checker.ts

```diff
@@ -5750,6 +5755,27 @@ namespace ts {
                 }
                 return Ternary.False;
             }
+
+            function enumRelatedTo(source: Type, target: Type) {
+                if (source.symbol.name !== target.symbol.name ||
+                    source.symbol.flags & SymbolFlags.ConstEnum ||
+                    target.symbol.flags & SymbolFlags.ConstEnum) {
+                    return Ternary.False;
+                }
+                const sourceDecl = <EnumDeclaration>getDeclarationOfKind(source.symbol, SyntaxKind.EnumDeclaration);

```

Comment:
what is means it that with this code
```ts
enum A { X }
enum A { Y }
```
you will see only `X` member, but you won't see `Y` since you are using one declaration to obtain property names.